specs/core/0_beacon-chain.md

```diff
@@ -695,15 +695,12 @@ def get_deposit_root() -> bytes32:
 @public
 @constant
 def get_merkle_branch(index: uint256) -> bytes32[32]: # size is DEPOSIT_CONTRACT_TREE_DEPTH (symbolic const not supported)
-    idx: uint256 = index + TWO_TO_POWER_OF_TREE_DEPTH
-    ret: bytes32[32] # size is DEPOSIT_CONTRACT_TREE_DEPTH
+    branch: bytes32[32] # size is DEPOSIT_CONTRACT_TREE_DEPTH
+    index = index + TWO_TO_POWER_OF_TREE_DEPTH
```

Comment:
This gives me a compilation error:
```
vyper.exceptions.ConstancyViolationException: Cannot modify function argument: index
```

Hence introducing the `idx` local.

Reply:
Sorry, no :-(

```
vyper.exceptions.FunctionDeclarationException: Variable name duplicated between function arguments and globals: deposit_count
```

While we're on naming, perhaps the method name should be just `get_branch()` - wdyt?

Reply:
We have compilation! 🎉 

I'll run it through the tests later to double check, but I can't imagine anything broke, so I'm happy for it to be merged as-is